newIDE/app/src/MainFrame/EditorContainers/HomePage/PlaySection/GamesPlatformFrameContext.js

```diff
@@ -310,59 +385,44 @@ const GamesPlatformFrameStateProvider = ({
   );
 
-  const sendTokenToIframeIfConnected = React.useCallback(
+  const { userCustomToken } = useUserCustomToken();
+
+  const sendUserCustomTokenToFrame = React.useCallback(
     async () => {
-      if (iframeLoaded && userId) {
-        // The iframe is loaded and the user is authenticated, so we can
-        // send that information to the iframe to automatically log the user in.
-        // $FlowFixMe - we know it's an iframe.
-        const iframe: ?HTMLIFrameElement = document.getElementById(
-          GAMES_PLATFORM_IFRAME_ID
-        );
-        if (iframe && iframe.contentWindow) {
-          try {
-            const userCustomToken = await generateCustomAuthToken(
-              getAuthorizationHeader,
-              userId
-            );
-            iframe.contentWindow.postMessage(
-              {
-                id: 'connectUserWithCustomToken',
-                token: userCustomToken,
-              },
-              // Specify the target origin to avoid leaking the customToken.
-              // Replace with '*' to test locally.
-              'https://gd.games'
-              // '*'
-            );
-          } catch (error) {
-            console.error(
-              'Error while generating custom token. User will not be logged in in the frame.',
-              error
-            );
-            return;
-          }
-        }
+      if (!iframeLoaded) {
+        return;
       }
-    },
-    [iframeLoaded, userId, getAuthorizationHeader]
-  );
 
-  React.useEffect(
-    () => {
-      sendTokenToIframeIfConnected();
-    },
-    [sendTokenToIframeIfConnected]
-  );
+      // The iframe is loaded:
+      // we can now sent the information that the user is connected,
+      // to automatically log the user in the frame,
+      // or notify it the user is not connected (or just disconnected).
 
-  const notifyIframeUserDisconnected = React.useCallback(
-    () => {
-      if (iframeLoaded && !userId) {
-        // $FlowFixMe - we know it's an iframe.
-        const iframe: ?HTMLIFrameElement = document.getElementById(
-          GAMES_PLATFORM_IFRAME_ID
-        );
-        if (iframe && iframe.contentWindow) {
+      // $FlowFixMe - we know it's an iframe.
+      const iframe: ?HTMLIFrameElement = document.getElementById(
+        GAMES_PLATFORM_IFRAME_ID
+      );
+      if (!iframe || !iframe.contentWindow) {
+        console.error('Iframe not found or not accessible.');
+        return;
+      }
+
+      try {
+        if (userCustomToken) {
+          console.log('Sending user custom token to Games Platform frame...');
+          iframe.contentWindow.postMessage(
+            {
+              id: 'connectUserWithCustomToken',
+              token: userCustomToken,
+            },
+            // Specify the target origin to avoid leaking the customToken.
+            // Use '*' to test locally.
+            Window.isDev() ? '*' : 'https://gd.games'
+          );
+        } else {
+          console.log(
```

Comment:
if you want to keep those console.log() maybe turn them into console.info() or remove